lib/components/viewers/favorite-stop-toggle.tsx

```diff
@@ -0,0 +1,80 @@
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import { Star as StarRegular } from '@styled-icons/fa-regular/Star'
+import { Star as StarSolid } from '@styled-icons/fa-solid/Star'
+import React, { useCallback } from 'react'
+
+import * as userActions from '../../actions/user'
+import { AppReduxState } from '../../util/state-types'
+import { getPersistenceMode } from '../../util/user'
+import { getShowUserSettings } from '../../util/state'
+import { StopData } from '../util/types'
+import { StyledIconWrapper } from '../util/styledIcon'
+
+interface Props {
+  enableFavoriteStops: boolean
+  forgetStop: (stopData: StopData) => void
+  isFavoriteStop: boolean
+  rememberStop: (stopData: StopData) => void
+  stopData?: StopData
+}
+
+/**
+ * Renders a clickable star to indicate/mark a stop as favorite.
+ */
+const FavoriteStopToggle = ({
+  enableFavoriteStops,
+  forgetStop,
+  isFavoriteStop,
+  rememberStop,
+  stopData
+}: Props): JSX.Element | null => {
+  const handleClick = useCallback(() => {
+    if (stopData) {
+      if (isFavoriteStop) forgetStop(stopData)
+      else rememberStop(stopData)
+    }
+  }, [forgetStop, isFavoriteStop, rememberStop, stopData])
+
+  return enableFavoriteStops ? (
+    <Button
+      bsSize="large"
+      bsStyle="link"
+      onClick={handleClick}
+      style={{
+        color: isFavoriteStop ? 'yellow' : 'black',
+        marginLeft: '5px',
+        padding: 0
+      }}
+    >
+      <StyledIconWrapper>
+        {isFavoriteStop ? <StarSolid /> : <StarRegular />}
+      </StyledIconWrapper>
+    </Button>
+  ) : null
+}
+
+// connect to redux store
+
+const mapStateToProps = (state: AppReduxState) => {
+  const showUserSettings = getShowUserSettings(state)
+  const { config, ui } = state.otp
+  const { persistence } = config
+  const stopId = ui.viewedStop.stopId
```

Comment:
Can you get the stop ID from the props rather than the state?

Reply:
The calculation might have to happen in the component

Reply:
Refactored in https://github.com/opentripplanner/otp-react-redux/pull/1153/commits/f568658c10bcd09ea5a4b70fc1d31db52c1a6a30.